cypress/integration/channel/mobile_message_deletion_spec.js

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getRandomInt} from '../../utils';
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/* eslint max-nested-callbacks: ["error", 4] */
+
+describe('Delete Parent Message', () => {
+    before(() => {
+        // 1. Go to Main Channel View with "user-1"
+        cy.viewport('iphone-6');
+        cy.toMainChannelView('user-1');
+    });
+
+    it('M14270 Deleting parent message should also delete replies from center and RHS', () => {
+        // 2. Close Hamburger menu, post a message, and add replies
+        cy.get('#post_textbox').click({force: true});
+        cy.postMessage('Parent Message');
+
+        cy.getLastPostId().then((postId) => {
+            cy.clickPostCommentIcon(postId);
+
+            // * Check that the RHS is open
+            cy.get('#rhsContainer').should('be.visible');
+
+            // * Add replies (randomly between 1 to 3)
+            const replyCount = getRandomInt(2) + 1;
+            for (var i = 0; i < replyCount; i++) {
+                cy.get('#reply_textbox').type('Reply').type('{enter}');
+
+                // add wait time to ensure that a post gets posted and not on pending state
+                cy.wait(500); // eslint-disable-line
```

Comment:
```
cy.wrap([...Array(replyCount)]).each(() => new Cypress.Promise((resolve) => {
  cy.get('#reply_textbox').type('Reply').type('{enter}').then(resolve);
}))
```

This will be awaited correctly.